projectq/ops/_gates.py

```diff
@@ -338,3 +338,49 @@ def get_inverse(self):
 
 #: Shortcut (instance of) :class:`projectq.ops.BarrierGate`
 Barrier = BarrierGate()
+
+
+class FlipBits(SelfInverseGate):
+    """ Gate for transforming qubits in state |0> to any computational basis state """
+    def __init__(self, bits_to_flip):
+        """
+        Initialize FlipBits gate.
+
+        Example:
+            .. code-block:: python
+
+                qureg = eng.allocate_qureg(2)
+                FlipBits([0, 1]) | qureg
+
+        Note:
+            The amplitude of state k is final_state[k]. When the state k is
+            written in binary notation, then qureg[0] denotes the qubit
+            whose state corresponds to the least significant bit of k.
+
+        Args:
+            bits_to_flip(list[int]|list[bool]|str): array of 0/1, True/False or
+                                    string of 0/1 identifying the qubits to flip
+                                    of length len(qureg).
+        """
+        if isinstance(bits_to_flip, str):
+            self.bits_to_flip = list([ c != "0" for c in bits_to_flip])
+        else:
+            self.bits_to_flip = list(bits_to_flip)
+
+    def __str__(self):
+        return "FlipBits"
+
+    def __or__(self, qubits):
+        for qureg in self.make_tuple_of_qureg(qubits):
+            for i, qubit in enumerate(qureg):
+                if self.bits_to_flip[i]:
+                    XGate() | qubit
+
+    def __eq__(self, other):
+        if isinstance(other, self.__class__):
+            return self.bits_to_flip == other.bits_to_flip
+        else:
```

Comment:
Done. Note that now that we allow an int to specify the bits to flip, it made sense to allow arrays that are shorter than `len(qureg)`. `__eq__()` will treat gates that flip the same bits but have `self.bits_to_flip` of different length (e.i., the differ by trailing `false` values) as not equal. I think this is the right behavior as these gates can be applied to `qureg`s of different length.

Reply:
Thanks! An alternative would be to store the integer instead of storing the list of bits (and then also compare integers). Furthermore, such an implementation would work as I'd expect for the input integer `-1`: It flips all qubits in the quantum register, no matter its length. What do you think?

Reply:
I am not sure this is the result I would "expect" that behavior, as for negative n, n-1, and not -n, has the complement of bits set compared with n, but I agree that it is a nice feature and have implemented this.